### ISSUE TITLE
Using variables from initialization during initialization

### ISSUE THREAD

**Issue by BartMitTyp** [samuelcolvin/pydantic#1226]
# Question

Output of `python -c "import pydantic.utils; print(pydantic.utils.version_info())"`:
```
pydantic version: 1.4
pydantic compiled: False
install path: C:\Users\name\Documents\v_env\Lib\site-packages\pydantic
python version: 3.7.3 (v3.7.3:ef4ec6ed12, Mar 25 2019, 22:22:05) [MSC v.1916 64 bit (AMD64)]
platform: Windows-10-10.0.14393-SP0
optional deps. installed: []
```

I would like to initialize a variable using variables I defined in during initialization. I had a list of objects previously, but simplified it for the issue. Am I using it the wrong way, or is this not yet possible?

```
from pydantic import BaseModel
from typing import List

class TestClass(BaseModel):
    x: int
    xs: List[int] = [x]
```
gets me the error
```
Traceback (most recent call last):
  File "C:/Users/name/.../tests.py", line 4, in <module>
    class TestClass(BaseModel):
  File "C:/Users/name/.../tests.py", line 6, in TestClass
    xs: List[int] = [x]
NameError: name 'x' is not defined
```

**Comment by samuelcolvin** [comment]
It's possible, you need to use validators, there's an examplen in the docs.

**Comment by samuelcolvin** [comment]
https://pydantic-docs.helpmanual.io/usage/validators/

See `passwords_match`.

**Comment by BartMitTyp** [comment]
Thank you for the quick answer. Is the desired way to set new variables actually this?
```
class TestClass(BaseModel):
    x: int

    @validator('x')
    def set_variables(cls, v, values):
        values['xs'] = [v]
        return v
```
It seems to work, but I thought there was a nicer way

**Comment by samuelcolvin** [comment]
That would work, but I was thinking more like

```py
from pydantic import BaseModel, validator
from typing import List

class TestClass(BaseModel):
    x: int
    xs: List[int] = None

    @validator('xs', always=True, pre=True)
    def set_cs(cls, v, values):
        return [values.get('x')]

debug(TestClass(x=4))
"""
test.py:14 <module>
    TestClass(x=4): TestClass(
        x=4,
        xs=[4],
    ) (TestClass)
"""
```

Once I fix #935 there should be a much more elegant solution than this, but that's the best approach for now.
